Add Typespec conversion option to intake template

mentat9 opened the pull request:
Intake form needs an option to indicate the author is converting swagger to Typespec. This change adds a checkbox for that option and indicates it should not be combined with adding a new API version.

# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

mentat9 committed:
Add Typespec converstion option to intake template

Intake form needs an option to indicate the author is converting swagger to Typespec. This change adds a checkbox for that option and indicates it should not be combined with adding a new API version.

mikeharder commented:
LGTM, but @konrad-jamrozik is best to approve.

konrad-jamrozik commented on .github/PULL_REQUEST_TEMPLATE/control_plane_template.md:
```diff
@@ -17,6 +17,7 @@ What's the purpose of this PR? Check the specific option that applies. This is *
   - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
   - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
   - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
+  - [ ] Convert existing swagger spec to Typespec (do not combine this with implementing a new API version).
```
```suggestion
  - [ ] Convert existing OpenAPI spec to TypeSpec spec (do not combine this with implementing a new API version).
```

We purposefully avoid using the term swagger as deprecated and incorrect. We agreed to this internally within Azure SDK plus there is this public blog post:

- https://swagger.io/blog/api-strategy/difference-between-swagger-and-openapi/

konrad-jamrozik replied:
Seems to me we should link here to some doc about TypeSpec conversions if we want to avoid unnecessary support burden.

I found this:

https://azure.github.io/typespec-azure/docs/next/howtos/migrate-swagger/get-started

Our TypeSpec page links to the page above from `Resources`, but that's it:

- [aka.ms/azsdk/typespec](https://aka.ms/azsdk/typespec)

I would not add this paragraph until we sort out the documentation, as I am usually one of the few people on first line of support fielding questions.

@mikeharder @MattGertz @rkmanda

konrad-jamrozik reviewed (changes requested):
See https://github.com/Azure/azure-rest-api-specs/pull/29616/files#r1661702412

mentat9 committed:
Update control_plane_template.md

Update text and add link to TypeSpec conversion docs per feedback.

mentat9 committed:
Update control_plane_template.md

mentat9 commented:
Updated the wording and added a link to docs for TypeSpec conversion. Also tweaked the comment about not combining TypeSpec conversion with API version changes, but it sounds like we haven't settled on the exact guidance. The last I heard, we were asking authors to NOT make an API version change as part of their TypeSpec conversion PR, but that may have changed?

Either way, from my perspective the important thing is that we don't want new APIs and feature changes included in the TypeSpec conversion PR. Open to input on the best wording for the guidance.

MattGertz commented:
@mentat9 It's correct that we don't want people to "version" APIs during the conversion -- i.e. change APIs in a functional way.  The current change to the file seems to state that correctly.  (Of course, the overall version number of the post-conversion, newly-deployed service will change since it is indeed an update, though one that ideally doesn't change anything functionally.  I wish we had better nouns for all of this.)

@konrad-jamrozik That document you suggest linking to looks correct, but looping in @allenjzhang to confirm this is the current official guidance for conversion.

mentat9 commented:
@MattGertz, @konrad-jamrozik - Note: I changed the doc link suggested by Konrad to point to its parent, since that seemed to be a little bit better landing page. Either one should be OK though. I'll change it if you feel strongly about it.

Commit:
Merge branch 'main' into mentat9-patch-1

konrad-jamrozik commented:
@mentat9 @MattGertz I sent out an email:

`Let's all agree on how to do API versions for TypeSpec conversion PRs: "a horizontal change that versions to preview"`

konrad-jamrozik commented:
Update: the email thread is ongoing and we are working on article to be hyperlinked from the template, which is [aka.ms/typespec/conversion](https://aka.ms/typespec/conversion).

mentat9 committed:
Update control_plane_template.md

Update help link for TypeSpec conversion

mentat9 commented:
> Update: the email thread is ongoing and we are working on article to be hyperlinked from the PR template, which is [aka.ms/typespec/conversion](https://aka.ms/typespec/conversion).

Updated the link.
